dao/mongodb.rya/src/test/java/org/apache/rya/mongodb/MongoDBRyaDAOTest.java

```diff
@@ -28,102 +30,279 @@
+
+        // Doc requires "A" and "B" and "C" and user has "A" and "B" and "C" = User can view
+        assertTrue(testVisibilityStatement("A&B&C", new Authorizations("A", "B", "C")));
+
+        // Doc requires "A" and "B" and "C" and user has "A" and "B" = User CANNOT view
+        assertFalse(testVisibilityStatement("A&B&C", new Authorizations("A", "B")));
+
+        // Doc requires "A" and "B" and user has "A" and "B" and "C" = User can view
+        assertTrue(testVisibilityStatement("A&B", new Authorizations("A", "B", "C")));
+
+        // Doc requires "A" or "B" and user has "A" = User can view
+        assertTrue(testVisibilityStatement("A|B", new Authorizations("A")));
+
+        // Doc requires "A" or "B" or "C" and user has "A" and "B" and "C" = User can view
+        assertTrue(testVisibilityStatement("A|B|C", new Authorizations("A", "B", "C")));
+
+        // Doc requires "A" or "B" or "C" and user has "A" and "B" = User can view
+        assertTrue(testVisibilityStatement("A|B|C", new Authorizations("A", "B")));
+
+        // Doc requires "A" or "B" or "C" and user has "A" = User can view
+        assertTrue(testVisibilityStatement("A|B|C", new Authorizations("A")));
+
+        // Doc requires "A" or "B" and user has "A" and "B" and "C" = User can view
+        assertTrue(testVisibilityStatement("A|B", new Authorizations("A", "B", "C")));
+
+        // Doc requires "A" and user has "" = User can view
+        assertTrue(testVisibilityStatement("A", Authorizations.EMPTY));
+
+        // Doc requires "A" and "B" and user has "" = User can view
+        assertTrue(testVisibilityStatement("A&B", Authorizations.EMPTY));
+
+        // Doc requires "A" or "B" and user has "" = User can view
+        assertTrue(testVisibilityStatement("A|B", Authorizations.EMPTY));
+
+        // Doc has no requirement and user has "" = User can view
+        assertTrue(testVisibilityStatement("", Authorizations.EMPTY));
+
+        // Doc has no requirement and user has "A" = User can view
+        assertTrue(testVisibilityStatement("", new Authorizations("A")));
+
+        // Doc has no requirement and user has "A" and "B" = User can view
+        assertTrue(testVisibilityStatement("", new Authorizations("A", "B")));
+
+        // Doc requires "A" or ("B" and "C") and user has "A" = User can view
+        assertTrue(testVisibilityStatement("A|(B&C)", new Authorizations("A")));
+
+        // Doc requires "A" or ("B" and "C") and user has "B" and "C" = User can view
+        assertTrue(testVisibilityStatement("A|(B&C)", new Authorizations("B", "C")));
+
+        // Doc requires "A" and ("B" or "C") and user has "A" and "B" = User can view
+        assertTrue(testVisibilityStatement("A&(B|C)", new Authorizations("A", "B")));
+
+        // Doc requires "A" and ("B" or "C") and user has "A" and "B" = User can view
+        assertTrue(testVisibilityStatement("A&(B|C)", new Authorizations("A", "C")));
```

Comment:
there are some missing assertFalse combinations that need to be added for completeness